<a href="https://colab.research.google.com/github/vohai2003/Pythoncode/blob/main/BT_TKMT_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Hệ số tương quan (Correlation) là hệ số đánh giá mối liên hệ giữa hai biến (hay hai cột dữ liệu - hai mẫu) \
Khoảng giá trị: [-1,1]. Nếu giá trị của hệ số rơi ra bên đoạn này, chứng tỏ phép tính lấy hệ số tương quan đã sai. \
Ý nghĩa của giá trị hệ số tương quan: \
p>0: Hai biến / mẫu đồng biến \
p < 0: Hai biến / mẫu nghịch biến \
p = 0: Hai biến / mẫu độc lập \

Ví dụ với tập dữ liệu về tỉ lệ tội phạm ở các bang ở Mĩ:

In [4]:
from google.colab import drive
drive.mount("/content/drive")
path = "/content/drive/My Drive/Dataset/"

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [6]:
import pandas
dt = pandas.read_csv(path+"state.csv")
processing_dtframe = pandas.DataFrame(columns = ['Population', 'Murder.Rate'], data = dt)
processing_dtframe.corr(method="pearson")

Population  Murder.Rate
Population     1.000000     0.182069
Murder.Rate    0.182069     1.000000

Đây là phương pháp Pearson để tính hệ số tương quan. \
Công thức tính: \
$p_{xy}=\frac{Cov(x,y)}{σ_xσ_y}$ \
Với tập dữ liệu trên, có thể thấy hệ số tương quan của hai đại lương: dân số (Population) và tỉ lệ tội phạm (Murder.Rate) có hệ số tương quan là 0.182069>0. Ta kết luận chúng đồng biến với nhau.